scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/GraphQLQueries.java

```diff
@@ -147,60 +147,179 @@ private JobInput getJobInput(List<String> statusNames, String user, List<Constra
         input.status(statusNames);
 
         String id = null;
+        String afterId = null;
+        String beforeId = null;
+        String status = null;
         String priority = null;
+        String userFilter = null;
         String name = null;
-        String statusString = null;
         String projectName = null;
+        long afterSubmittedTime = -1;
+        long beforeSubmittedTime = -1;
 
         for (Constraint constraint : constraints) {
             String value = constraint.getValue();
             switch (constraint.getTargetField()) {
                 case ID: {
+
                     switch (constraint.getAction()) {
                         case EQUALS:
-                            id = getValue(id, value);
+                            if (value != null) {
+                                if (id == null)
+                                    id = value;
+                                else if (value != id)
+                                    return input.jobName(RETURN_NOTHING_FILTER).build();
+                            }
                             break;
                         case GREATER_THAN_OR_EQUAL_TO:
-                            input.afterId(value);
+                            if (value != null && (afterId == null || Integer.valueOf(value) > Integer.valueOf(afterId)))
+                                afterId = value;
                             break;
                         case LESS_THAN_OR_EQUAL_TO:
-                            input.beforeId(value);
+                            if (value != null &&
+                                (beforeId == null || Integer.valueOf(value) < Integer.valueOf(beforeId)))
```

Comment:
and if I give a string value? an exception will be triggered by the Integer.valueOf(value).

Reply:
You re right. It should filter nothing rather than throwing a validation type error.. I am fixing this.